tests/test_modeling_qdqbert.py

```diff
@@ -0,0 +1,549 @@
+
+    def test_for_causal_lm_decoder(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs_for_decoder()
+        self.model_tester.create_and_check_model_for_causal_lm_as_decoder(*config_and_inputs)
+
+    def test_decoder_model_past_with_large_inputs(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs_for_decoder()
+        self.model_tester.create_and_check_decoder_model_past_large_inputs(*config_and_inputs)
+
+    def test_for_multiple_choice(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_for_multiple_choice(*config_and_inputs)
+
+    def test_for_next_sequence_prediction(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_for_next_sequence_prediction(*config_and_inputs)
+
+    def test_for_question_answering(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_for_question_answering(*config_and_inputs)
+
+    def test_for_sequence_classification(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_for_sequence_classification(*config_and_inputs)
+
+    def test_for_token_classification(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_for_token_classification(*config_and_inputs)
+
+    @slow
+    def test_model_from_pretrained(self):
+        for model_name in QDQBERT_PRETRAINED_MODEL_ARCHIVE_LIST[:1]:
+            model = QDQBertModel.from_pretrained(model_name)
+            self.assertIsNotNone(model)
+
+    # Override
+    def test_feed_forward_chunking(self):
+        pass
+
+
+@require_torch
+@require_quantization
+class QDQBertModelIntegrationTest(unittest.TestCase):
+    @slow
+    def test_inference_no_head_absolute_embedding(self):
+
+        import pytorch_quantization.nn as quant_nn
+        from pytorch_quantization.tensor_quant import QuantDescriptor
+
+        model = QDQBertForMaskedLM.from_pretrained("bert-base-uncased")
+        input_ids = torch.tensor([[0, 345, 232, 328, 740, 140, 1695, 69, 6078, 1588, 2]])
+
+        input_desc = QuantDescriptor(num_bits=8, calib_method="max")
+        weight_desc = QuantDescriptor(num_bits=8, axis=((0,)))
+        quant_nn.QuantLinear.set_default_quant_desc_input(input_desc)
+        quant_nn.QuantLinear.set_default_quant_desc_weight(weight_desc)
+
+        output = model(input_ids)[0]
+
+        expected_shape = torch.Size((1, 11, 768))
+        self.assertEqual(output.shape, expected_shape)
+
+        expected_slice = torch.tensor(
+            [[[-0.0483, 0.1188, -0.0313], [-0.0606, 0.1435, 0.0199], [-0.0235, 0.1519, 0.0175]]]
+        )
+        self.assertTrue(torch.allclose(output[:, :3, :3], expected_slice, atol=1e-4))
```

Comment:
If this approach is the canonical approach to using the model, I would favor updating the model docstring examples to reflect that, alongside with inline comments explaining in a few words what is happening. I think this would go a long way to improve usability of the model. Thank you!

Reply:
addressed in https://github.com/huggingface/transformers/pull/14066/commits/69da5d6f7c91ecc219d9f4560930fd61689c8435